packages/xod-project/src/flatten.js

```diff
@@ -854,26 +888,30 @@ const convertPatch = def(
  */
 const convertProject = def(
   'convertProject :: Project -> PatchPath -> Patch -> [Pair PatchPath Patch] -> Either Error Project',
-  (project, path, patch, leafPatches) => {
-    // Patch
-    const convertedPatch = convertPatch(project, leafPatches, patch);
-    // String[]
-    const usedCastNodeTypes = getCastNodeTypesFromPatch(convertedPatch);
-    // Right Project
-    const newProject = Either.of(Project.createProject());
-
-    return R.compose(
-      R.chain(reduceChainOver(newProject)),
-      R.map(R.compose(
-        R.map(R.apply(Project.assocPatch)),
-        R.append([path, convertedPatch]),
-        removeNotImplementedInXodNodes,
-        removeTerminalPatches
-      )),
-      R.sequence(Either.of),
-      addCastPatches(project, usedCastNodeTypes)
-    )(leafPatches);
-  }
+  (project, path, patch, leafPatches) => R.compose(
+    composeDeadRefError(path),
```

Comment:
The `composeDeadRefError` is misguiding. Sounds like we always create an error here rather than qualifying the existing.

```js
foldEither(
  err =>  Either.Left(assocPatchPathToError(path, err)),
  R.identity
)
```